frontend/src/components/DropDownGroup.jsx

```diff
@@ -0,0 +1,205 @@
+import { Menu } from "antd";
+import { useState, useEffect } from "react";
+import { NavLink } from "react-router-dom";
+import { useContext } from "react";
+import { AuthContext } from "../components/AuthContext";
+import Logout from "../components/Logout";
+
+const DropDownGroup = () => {
+  const [current, setCurrent] = useState("");
+  const { user, loading } = useContext(AuthContext);
+  const [links, setLinks] = useState();
+
+  const onClick = (e) => {
+    setCurrent(e.key);
+  };
+
+  const userCheckHours = () => {
+    if (!user) {
+      return null;
+    } else {
+      try {
+        const parsedClaims = JSON.parse(
+          user.proactiveRefresh.user.reloadUserInfo.customAttributes
+        );
+        const parsedPalkka = parsedClaims["Kaikki tunnit"];
+        return parsedPalkka;
+      } catch (error) {
+        return null;
+      }
+    }
+  };
+  const userCheckCumulativeHours = () => {
+    if (!user) {
+      return null;
+    } else {
+      try {
+        const parsedClaims = JSON.parse(
+          user.proactiveRefresh.user.reloadUserInfo.customAttributes
+        );
+        const parsedPalkka = parsedClaims["Kaikki tunnit kumulatiivinen"];
+        return parsedPalkka;
+      } catch (error) {
+        return null;
+      }
+    }
+  };
+  const userCheckSalary = () => {
+    if (!user) {
+      return null;
+    } else {
+      try {
+        const parsedClaims = JSON.parse(
+          user.proactiveRefresh.user.reloadUserInfo.customAttributes
+        );
+        const parsedPalkka = parsedClaims["Palkka"];
+        return parsedPalkka;
+      } catch (error) {
+        return null;
+      }
+    }
+  };
+  const userCheckBillability = () => {
+    if (!user) {
+      return null;
+    } else {
+      try {
+        const parsedClaims = JSON.parse(
+          user.proactiveRefresh.user.reloadUserInfo.customAttributes
+        );
+        const parsedPalkka = parsedClaims["Laskutettavat tunnit"];
+        return parsedPalkka;
+      } catch (error) {
+        return null;
+      }
+    }
+  };
+
+  const userCheckAdmin = () => {
+    if (!user) {
+      return null;
+    } else {
+      try {
+        const parsedClaims = JSON.parse(
+          user.proactiveRefresh.user.reloadUserInfo.customAttributes
+        );
+        const parsedAdmin = parsedClaims["admin"];
+        return parsedAdmin;
+      } catch (error) {
+        return null;
+      }
+    }
+    //
+  };
+
+  const items = [];
+  const itemsConstructor = () => {
+    if (userCheckHours() || userCheckCumulativeHours() || userCheckSalary()) {
+      items.push({
+        label: "Henkilöstö",
+        key: "SubMenu1",
+        children: [],
+      });
+    } else {
+      items.push({});
+    }
+    if (userCheckHours()) {
+      items[0].children.push({
```

Comment:
Instead of using items[some_index] to access menu items when pushing children, you can store the menu item in a variable (you can call it for example `staff` and then add the child elements and after they are all added then add the variable to items.

const staff = {label: "Henkilöstö", .......} and then staff.children.push for individiual links, finally use items.push(staff).